### PR TITLE
Fix issue #3686

### PR DESCRIPTION
**Pull request by iRmantou**

### CODE REVIEW

File: src/shogun/labels/BinaryLabels.cpp [hunk 1]
```diff
@@ -64,6 +65,22 @@ CBinaryLabels::CBinaryLabels(CFile * loader) : CDenseLabels(loader)
 {
 }
 
+CBinaryLabels::operator CMulticlassLabels()
+{
+	CMulticlassLabels	src;
+	
+	for (int32_t i = 0; i < m_labels.vlen; i++)
```
Comment (iRmantou) [inline src/shogun/labels/BinaryLabels.cpp]: Ok, I have fixed them! : )

File: src/shogun/labels/BinaryLabels.cpp [hunk 2]
```diff
@@ -64,6 +65,22 @@ CBinaryLabels::CBinaryLabels(CFile * loader) : CDenseLabels(loader)
 {
 }
 
+CBinaryLabels::operator CMulticlassLabels()
+{
+	CMulticlassLabels src;
+	
+	for (index_t i = 0; i < m_labels.vlen; i++)
+	{
+		if(m_labels.vector[i] == -1)
```
Comment (karlnapf) [inline src/shogun/labels/BinaryLabels.cpp]: this is not the correct conversion

Comment (karlnapf) [inline src/shogun/labels/BinaryLabels.cpp]: Actually, sorry, it is in most cases.
But this needs checks and user readable error messages

File: src/shogun/machine/MulticlassMachine.cpp [hunk 3]
```diff
@@ -51,7 +51,7 @@ CMulticlassMachine::~CMulticlassMachine()
 
 void CMulticlassMachine::set_labels(CLabels* lab)
 {
-    CMachine::set_labels(lab);
+    CMachine::set_labels(dynamic_cast<CMulticlassLabels*>(lab));
```
Comment (karlnapf) [inline src/shogun/machine/MulticlassMachine.cpp]: unit test pls

File: src/shogun/labels/BinaryLabels.cpp (hunk 2 above)
Comment (karlnapf) [inline src/shogun/labels/BinaryLabels.cpp]: space after if

File: src/shogun/labels/BinaryLabels.cpp [hunk 4]
```diff
@@ -64,6 +65,22 @@ CBinaryLabels::CBinaryLabels(CFile * loader) : CDenseLabels(loader)
...
+		{
```
Comment (karlnapf) [inline src/shogun/labels/BinaryLabels.cpp]: no {} for single line commands

Comment (iRmantou) [inline src/shogun/labels/BinaryLabels.cpp]: Ok, I will fix these style problems, but I saw some source code that use {} for single line commands in BinaryLabels.cpp, so I Imitate them.

File: src/shogun/machine/MulticlassMachine.cpp (hunk 3 above)
Comment (iRmantou) [inline src/shogun/machine/MulticlassMachine.cpp]: I have run `shogun-unit-test` before I sent this pr, I am sorry I dont know where I am wrong, you mean I need create a small test or example passing `CBinaryLabels ` to `CMulticlassClassifier`?

**Comment by vigsterkr** [comment]
@iRmantou i think you mixed up a commit in this pr... it's great that you've tried to fix the `KernelMachine` and use openmp, but do that in a separate pr.

**Comment by iRmantou** [comment]
@vigsterkr  yes this is a mistake... I will sent a new pr...and delete this... I am a little tired.. sorry

**Comment by vigsterkr** [comment]
@iRmantou for the future reference... you should just force push stuff into your branch to reset the pr... no need to open a brand new pr...

**Comment by iRmantou** [comment]
@vigsterkr I have created a new branch and sent a new pr fixing the `KernelMachine` you can see it in #3605 ,and I have removed the wrong commits in this pr to make it clear. Could you reopen this pr, plz? I think github will refresh this page. thank u!

**Comment by iRmantou** [comment]
@vigsterkr I have created a new branch and sent a new pr fixing the `KernelMachine` ,and I have removed the wrong commits in this pr to make it clear. Could you reopen this pr, plz? I think github will refresh this page. thank u!

**Pull request closed without merging**